flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveScheduler.java

```diff
@@ -785,18 +837,25 @@ public ArchivedExecutionGraph getArchivedExecutionGraph(
 
     @Override
     public void goToWaitingForResources() {
-        final ResourceCounter desiredResources = calculateDesiredResources();
-        declarativeSlotPool.setResourceRequirements(desiredResources);
+        declareDesiredResources();
 
         transitionToState(
                 new WaitingForResources.Factory(
```

Comment:
Is it correct that we only enter this state in case of initial job start or when we lose a required slot (e.g. loss of a task manager)? In all other cases we should be allocating new slots first (if required) while the job keeps running.

Reply:
Yes, we only enter the state on the job start or during the job restart (restart happens in case of failure or when we decide to rescale - see Executing#checkChangingParallelism).